### PR TITLE
Iox #398 fix bug in run_all_tests.sh

### LINKED ISSUE #398: run_all_tests.sh wont run all the tests
## Required information

**Operating system:**
Ubuntu / Mac 

**Compiler version:**
NA

**Observed result or behaviour:**
run_all_tests.sh runs only `<comp>_moduletests` 

**Expected result or behaviour:**
What do you expect to happen?
run_all_tests.sh should run other tests as well e.g. `<comp>_integrationtests` & `<comp>_componenttests`

**Conditions where it occurred / Performed steps:**

- Build iceoryx using `./tools/iceoryx_build_test.sh build-test`
- Step into build folder `cd build`
- Run tests `./tools/run_all_tests.sh`

### PR DESCRIPTION
**Pull request by prasannabhat**
Signed-off-by: Prasanna Bhat <prasanna.yoga@gmail.com>

## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)

1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Notes for Reviewer

`;&` operator enables fall through in case statement in shell script.
[This operator](https://stackoverflow.com/questions/12010686/case-statement-fallthrough) was introduced in bash 4.0.
**NOTE :** This change is reverted ([wont work in Mac](https://github.com/eclipse/iceoryx/runs/1467255289)) and `if`.statement is used instead

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [x] All checkboxes in the PR checklist are checked or crossed out
1. [x] Merge

## References

- Closes #398

### CODE REVIEW

File: tools/run_all_tests.sh [hunk 1]
```diff
@@ -68,24 +68,53 @@ echo ">>>>>> Running Ice0ryx Tests <<<<<<"
 
 set -e
 
+execute_test () {
+    local component=$1
+    local test_scope=$2
+
+    case $test_scope in
+        "unit")
+        test_binary="$component"_moduletests
```
Comment (ithier) [inline tools/run_all_tests.sh:77]: Variables should be `ALL_CAPS` according to the existing convention in this file.

Comment (ithier) [inline tools/run_all_tests.sh:77]: Variables should be `ALL_CAPS` according to the existing convention in this script.

**Review by ithier — CHANGES_REQUESTED** [review]

File: tools/run_all_tests.sh (hunk 1 above)
Comment (prasannabhat) [reply tools/run_all_tests.sh:78]: I changed the scope of the variable to function local.
I guess we can use small caps for local variables (better readable)

Comment (ithier) [reply tools/run_all_tests.sh:78]: Makes sense

**Review by elBoberido — APPROVED** [review]
I'm not a big bash hacker, but this looks good to me. Maybe @dkroenke can also have a look at this.

Additionally, @elfenpiff told me he fixed some issues with the test in his waitset PR ... yes, it seems it was not big enough for him 😁. Maybe he can also check this and revert his changes

**Review by budrus — APPROVED** [review]

**Review by ithier — APPROVED** [review]

**Pull request merged**